tests/integration_tests/test_mariadb.py

```diff
@@ -109,9 +115,180 @@ def setUpClass(cls):
+            select
+                rental_price, location, sqft, $rental_price_confidence, number_of_rooms
+            from
+                mindsdb.{test_predictor_name} where $select_data_query='select * from test.{test_data_table} limit 3';
+        """, as_dict=True)
+
+        print('check result')
+        self.assertTrue(len(results) == 3)
+        for res in results:
+            self.assertTrue(res['rental_price'] is not None and res['rental_price'] != 'None')
+            self.assertTrue(res['location'] is not None and res['location'] != 'None')
+            self.assertIsInstance(res['$rental_price_confidence'], float)
+
+    def test_5_delete_predictor_by_command(self):
+        print(f'\nExecuting {inspect.stack()[0].function}')
+
+        query(f"""
+            insert into mindsdb.commands values ('delete predictor {test_predictor_name}');
+        """)
+
+        print(f'Predictor {test_predictor_name} not exists')
+        models = [x['name'] for x in self.mdb.get_models()]
+        self.assertTrue(test_predictor_name not in models)
+
+        print(f'Test predictor table not exists')
+        mindsdb_tables = query('show tables from mindsdb')
+        mindsdb_tables = [x[0] for x in mindsdb_tables]
+        self.assertTrue(test_predictor_name not in mindsdb_tables)
+
+    def test_6_insert_predictor_again(self):
+        print(f'\nExecuting {inspect.stack()[0].function}')
+        self.test_2_insert_predictor()
+
+    def test_7_delete_predictor_by_delete_statement(self):
+        print(f'\nExecuting {inspect.stack()[0].function}')
+        query(f"""
+            delete from mindsdb.predictors where name='{test_predictor_name}';
+        """)
+
+        print(f'Predictor {test_predictor_name} not exists')
+        models = [x['name'] for x in self.mdb.get_models()]
+        self.assertTrue(test_predictor_name not in models)
+
+        print(f'Test predictor table not exists')
+        mindsdb_tables = query('show tables from mindsdb')
+        mindsdb_tables = [x[0] for x in mindsdb_tables]
+        self.assertTrue(test_predictor_name not in mindsdb_tables)
+
+
+def wait_mysql(timeout):
+    global config
+    m = Mariadb(config)
+
+    start_time = time.time()
+
+    connected = m.check_connection()
+    while connected is False and (time.time() - start_time) < timeout:
+        time.sleep(2)
+        connected = m.check_connection()
+
+    return connected
+
+def is_port_in_use(port_num):
+    portsinuse = []
+    conns = psutil.net_connections()
+    portsinuse = [x.laddr[1] for x in conns if x.status == 'LISTEN']
+    portsinuse.sort()
+    return int(port_num) in portsinuse
+
+def wait_port(port_num, timeout):
+    start_time = time.time()
+
+    in_use = is_port_in_use(port_num)
+    while in_use is False and (time.time() - start_time) < timeout:
+        time.sleep(2)
+        in_use = is_port_in_use(port_num)
+
+    return in_use
+
 
 if __name__ == "__main__":
-    unittest.main()
+    maria_sp = subprocess.Popen(
```

Comment:
Is running docker-compose from python necessary in this case? Seems like some very complicated logic, but it's really solved by having the user just run `./cli.sh` on their own and waiting for a bit

Reply:
Cant agree in this case.  Is much easy make one action (run test file) instead of three (start container and wait it ready, run test file, stop container)